tests/test_ranges.py

```diff
@@ -41,272 +41,265 @@ def valid_integer_range(draw):
-
-
-@given(valid_integer_range(), valid_integer_range())
-def test_range_is_disjoint(a: ranges.Range[Any], b: ranges.Range[Any]) -> None:
-    assert a.intersection(b) is not None or a.is_disjoint(b)
-
-
-@pytest.mark.parametrize(
-    "a_str,b_str,expected_str",
-    [
-        # Finite intersections
-        ("[0,2)", "[1,3)", "[1,2)"),
-        ("[0,2]", "[1,3)", "[1,2]"),
-        # Interactions with unbounded ranges
-        ("[0,None)", "[1,3)", "[1,3)"),
-        ("[0,None)", "[2,None)", "[2,None)"),
-        ("[0,3)", "[2,None)", "[2,3)"),
-        ("(None,3)", "[0,4)", "[0,3)"),
-        ("(None,3)", "[0,None)", "[0,3)"),
-        # Disjoint ranges
-        ("[0,2)", "[2,4)", None),
-        ("[0,2)", "[3,4)", None),
-        ("(None,2)", "[2,None)", None),
-        ("(None,2)", "[3,None)", None),
-        # Other range types
-        ("(0,2)", "(1,3)", "(1,2)"),
-        ("(0,2)", "(2,4)", None),
-        ("(0,2]", "(1,3]", "(1,2]"),
-        ("(0,2]", "(2,4]", None),
-        ("[0,2]", "[1,3]", "[1,2]"),
-        ("[0,2]", "[2,4]", "[2,2]"),
-        # Mixed range types
-        ("(0,2)", "[1,3)", "[1,2)"),
-        ("(0,2)", "[-1,1)", "(0,1)"),
-        ("(0,3)", "[1,2)", "[1,2)"),
-        ("(0,2)", "[2,3)", None),
-        ("[0,2]", "(0,2)", "(0,2)"),
-        ("(0,2]", "[2,4)", "[2,2]"),
-    ],
-)
-def test_intersection(a_str, b_str, expected_str):
-    a = _range_from_string(a_str)
-    b = _range_from_string(b_str)
-    if expected_str is not None:
-        expected = _range_from_string(expected_str)
-    else:
-        expected = None
-
-    result = a & b
-
-    assert result == expected
-
-
-@pytest.mark.parametrize(
-    "a_str,b_str,expected_str",
-    [
-        # Finite intersections
-        ("[0,2)", "[1,3)", "[0,3)"),
-        ("[1,3)", "[0,2)", "[0,3)"),
-        # Interactions with unbounded ranges
-        ("[1,None)", "[0,3)", "[0,None)"),
-        ("[0,None)", "[2,None)", "[0,None)"),
-        ("(None,3)", "[0,4)", "(None,4)"),
-        ("(None,3)", "[0,None)", "(None,None)"),
-        # Adjacent ranges
-        ("[0,2)", "[2,4)", "[0,4)"),
-        ("[0,2]", "(2,4)", "[0,4)"),
-        # Disjoint ranges
-        ("[0,2)", "[3,4)", None),
-        ("(None,2)", "[3,None)", None),
-        # Other range types
-        ("(0,2)", "(1,3)", "(0,3)"),
-        ("(0,2)", "(2,4)", None),
-        ("(0,2]", "(1,3]", "(0,3]"),
-        ("(0,2]", "(2,4]", "(0,4]"),
-        ("[0,2]", "[1,3]", "[0,3]"),
-        ("[0,2]", "[2,4]", "[0,4]"),
-        # Mixed range types
-        ("(0,2)", "[0,2)", "[0,2)"),
-        ("(0,2)", "[2,3)", "(0,3)"),
-        ("[0,2]", "(0,2)", "[0,2]"),
-        ("[0,2]", "(0,3)", "[0,3)"),
-        ("(0,2]", "[2,4)", "(0,4)"),
-    ],
-)
-def test_range_union(a_str, b_str, expected_str):
-    a = _range_from_string(a_str)
-    b = _range_from_string(b_str)
-    if expected_str is not None:
-        expected = _range_from_string(expected_str)
-    else:
-        expected = None
+class TestRange:
```

Comment:
🧀 This is a awkward diff but the only change I'm making is just:

- Creating a new `TestX` class for the class of functions under test.
- Indenting the existing test code to be within these classes.
- Adding a `self` parameter to each encapsulated test function.

The only existing to this is `test_finite_range` (see below).